src/Solnet.Extensions/Models/TokenWallet/TokenWalletFilterList.cs

```diff
@@ -0,0 +1,152 @@
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return _list.GetEnumerator();
+        }
+
+        /// <summary>
+        /// Keeps all accounts that match the TokenDef provided.
+        /// </summary>
+        /// <param name="token">An instance of TokenDef to use for filtering.</param>
+        /// <returns>A filtered list of accounts that match the supplied TokenDef.</returns>
+        public TokenWalletFilterList ForToken(TokenDef token)
+        {
+            if (token == null) throw new ArgumentNullException(nameof(token));
+            return new TokenWalletFilterList(_list.Where(x => x.TokenMint == token.TokenMint));
+        }
+
+        /// <summary>
+        /// Keeps all accounts with the token symbol supplied.
+        /// <para>Be aware that token symbol does not guarentee you are interacting with the TokenMint you think. 
+        /// It is much safer to identify tokens using their token mint public key address.</para>
+        /// </summary>
+        /// <param name="symbol">A token symbol, e.g. USDC</param>
+        /// <returns>A filtered list of accounts for the given token symbol.</returns>
+        public TokenWalletFilterList WithSymbol(string symbol)
+        {
+            if (string.IsNullOrWhiteSpace(symbol)) throw new ArgumentException(nameof(symbol));
+            return new TokenWalletFilterList(_list.Where(x => x.Symbol == symbol));
+        }
+
+        /// <summary>
+        /// Get the TokenWalletAccount for the public key provided.
+        /// </summary>
+        /// <param name="publicKey">Public key for the account</param>
+        /// <returns>The account with the matching public key or null if not found.</returns>
+        public TokenWalletAccount WithPublicKey(string publicKey)
+        {
+            if (string.IsNullOrWhiteSpace(publicKey)) throw new ArgumentException(nameof(publicKey));
+            return new TokenWalletFilterList(_list.Where(x => x.PublicKey == publicKey)).FirstOrDefault();
+        }
+
+        /// <summary>
+        /// Keeps all accounts for the given token mint address.
+        /// </summary>
+        /// <param name="mint">Token mint public key address.</param>
+        /// <returns>A filtered list of accounts for the given mint.</returns>
+        public TokenWalletFilterList WithMint(string mint)
+        {
+            return new TokenWalletFilterList(_list.Where(x => x.TokenMint == mint));
+        }
+
+        /// <summary>
+        /// Keeps all accounts with at least the supplied minimum balance.
+        /// </summary>
+        /// <param name="minimumBalance">A minimum balance value as decimal.</param>
+        /// <returns>A filtered list of accounts with at least the balance as decimal supplied.</returns>
+        public TokenWalletFilterList WithAtLeast(decimal minimumBalance)
+        {
+            return new TokenWalletFilterList(_list.Where(x => x.QuantityDecimal >= minimumBalance));
+        }
+
```

Comment:
I would add a variant of this filter for > 0

`public TokenWalletFilterList WithNonZero()`

Would be very useful in a wallet UI to display tokens with any value :)